vendors/circom/circomlib/zkey/coefficient.h

```diff
@@ -0,0 +1,53 @@
+#ifndef VENDORS_CIRCOM_CIRCOMLIB_ZKEY_COEFFICIENT_H_
+#define VENDORS_CIRCOM_CIRCOMLIB_ZKEY_COEFFICIENT_H_
+
+#include <stdint.h>
+
+#include <string>
+
+#include "absl/strings/substitute.h"
+
+namespace tachyon::circom {
+
+#pragma pack(push, 1)
+// R1CS is represented as A * z ∘ B * z = C * z, where ∘ is the Hadamard
+// product. Each constraint is composed as follows:
+//
+// - [aᵢ,₀, ...,  aᵢ,ₘ₋₁] * [z₀, ..., zₘ₋₁]
+// - [bᵢ,₀, ...,  bᵢ,ₘ₋₁] * [z₀, ..., zₘ₋₁]
+// - [cᵢ,₀, ...,  cᵢ,ₘ₋₁] * [z₀, ..., zₘ₋₁]
+//
+// where i is the index of the constraints (0 ≤ i < n),
+// m is the number of QAP variables, and n is the number of constraints.
+//
+// The last one is computed if we know the first two ones.
+// Therefore, |Coefficient| represents the first two ones.
+template <typename F>
+struct Coefficient {
+  // if |matrix| is 0, then this constraint is for A matrix, otherwise B matrix.
```

Comment:
To follow the standard of the other variable comments:
```suggestion
  // A value denoting the matrix this constraint is for. If 0, this constraint is for matrix. Else, this constraint is for matrix B.